pages/api/v1/delegate-chat/getChatMessages.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import { IMessage } from '~src/types';
+import getEncodedAddress from '~src/util/getEncodedAddress';
+import { isAddress } from 'ethers';
+import getTokenFromReq from '~src/auth/utils/getTokenFromReq';
+import { MessageType } from '~src/auth/types';
+import messages from '~src/auth/utils/messages';
+import authServiceInstance from '~src/auth/auth';
+import storeApiKeyUsage from '~src/api-middlewares/storeApiKeyUsage';
+import { chatDocRef, chatMessagesRef } from '~src/api-utils/firestore_refs';
+import getSubstrateAddress from '~src/util/getSubstrateAddress';
+
+async function handler(req: NextApiRequest, res: NextApiResponse<{ messages: IMessage[] } | MessageType>) {
+	storeApiKeyUsage(req);
+
+	const network = String(req.headers['x-network']);
+	if (!network || !isValidNetwork(network)) return res.status(400).json({ message: 'Invalid network in request header' });
+
+	const token = getTokenFromReq(req);
+	if (!token) return res.status(401).json({ message: 'Invalid token' });
+
+	const user = await authServiceInstance.GetUser(token);
+	if (!user) return res.status(403).json({ message: messages.UNAUTHORISED });
+
+	const { chatId, address } = req.body;
+	if (!chatId || !address || typeof chatId !== 'string' || chatId.trim() === '') return res.status(400).json({ message: messages.INVALID_PARAMS });
+	if (!(getEncodedAddress(String(address), network) || isAddress(String(address)))) return res.status(400).json({ message: 'Invalid address' });
+
+	const substrateAddress = getSubstrateAddress(address);
+	try {
+		const chatDoc = await chatDocRef(chatId).get();
```

Comment:
please add a check for chatDoc.exists before checking for data, it might throw an null error